server/graphql/v1/mutations/notifications.js

```diff
@@ -0,0 +1,110 @@
+import Promise from 'bluebird';
+import { values } from 'lodash';
+import config from 'config';
+import models, { Op } from '../../../models';
+import { activities } from '../../../constants';
+import { channels } from '../../../constants';
+import { Forbidden, NotFound, Unauthorized } from '../../errors';
+
+const NotificationPermissionError = new Forbidden({
+  message: "This notification does not exist or you don't have the permission to edit it.",
+});
+
+const MaxWebhooksExceededError = new Forbidden({
+  message: 'You have reached the webhooks limit for this collective.',
+});
+
+/**
+ * Edits (by replacing) the admin-level webhooks for a collective.
+ */
+export async function editWebhooks(args, remoteUser) {
+  if (!(remoteUser && remoteUser.isAdmin(args.collectiveId))) {
+    throw NotificationPermissionError;
+  }
+
+  if (!args.notifications) return Promise.resolve();
+
+  const oldNotifications = await models.Notification.findAll({
+    where: { CollectiveId: args.collectiveId, UserId: null, channel: channels.WEBHOOK },
+  });
+
+  const toDelete = oldNotifications
+    .filter(
+      oldNotification =>
+        !args.notifications.some(newNotification => {
+          return (
+            newNotification.type === oldNotification.type && newNotification.webhookUrl === oldNotification.webhookUrl
+          );
+        }),
+    )
+    .map(x => x.id);
+
+  const toCreate = args.notifications.filter(
+    newNotification =>
+      !oldNotifications.some(oldNotification => {
+        return (
+          oldNotification.type === newNotification.type && oldNotification.webhookUrl === newNotification.webhookUrl
+        );
+      }),
+  );
+
+  models.Notification.destroy({ where: { id: { [Op.in]: toDelete } } });
+
+  return Promise.map(toCreate, notification => {
+    if (!(values(activities).includes(notification.type) && notification.channel === channels.WEBHOOK)) return;
+
+    notification.CollectiveId = args.collectiveId;
+    return models.Notification.create(notification);
+  });
+}
+
+/**
+ * Creates a Webhook subscription for a collective given a collective slug.
+ */
+export async function createWebhook(args, remoteUser) {
+  if (!(remoteUser && remoteUser.isAdmin(args.collectiveId))) {
```

Comment:
I think `createWebhook` is taking a `collectiveSlug` (not a `collectiveId`), does that work?

Reply:
It doesn't. I've fixed it.